### PR TITLE
Set up NavigationTabs and NavigationTabItem

### PR DESCRIPTION
**Pull request by beaesguerra**
## Summary:
These changes set up the `wonder-blocks-tabs` package and the `NavigationTabs` and `NavigationTabItem` components.

So far, these components implement the html semantics and base styles. Will be implementing the props and addressing more styling edge cases in upcoming PRs.

Issue: WB-1898

## Test plan:
Review the docs for:
- NavigationTabs (`?path=/docs/packages-tabs-navigationtabs-navigationtabs--docs`)
- NavigationTabItem (`?path=/docs/packages-tabs-navigationtabs-navigationtabitem--docs`)

Review base styling for the components

### CODE REVIEW

File: __docs__/wonder-blocks-tabs/navigation-tab-item.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    },
+} as Meta<typeof NavigationTabItem>;
+
+type StoryComponentType = StoryObj<typeof NavigationTabItem>;
+
+export const Default: StoryComponentType = {
+    args: {
+        children: <Link href="#link">Navigation tab item</Link>,
+    },
+};
```
Comment (jandrade) [inline __docs__/wonder-blocks-tabs/navigation-tab-item.stories.tsx:41]: **suggestion:** It seems that this case would be already covered by `All Variants`, so you could disable the chromatic snapshot.

File: __docs__/wonder-blocks-tabs/navigation-tabs.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    <NavigationTabItem>
+        <Link href="#link-3">Navigation tab item 3</Link>
+    </NavigationTabItem>,
+];
+
+export const Default: StoryComponentType = {
+    args: {
+        children: navigationTabItems,
+    },
+};
```
Comment (jandrade) [inline __docs__/wonder-blocks-tabs/navigation-tabs.stories.tsx:49]: **suggestion:** It seems that this case would be already covered by `All Variants`, so you could disable the chromatic snapshot.

File: __docs__/wonder-blocks-tabs/navigation-tab-item.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import type {Meta, StoryObj} from "@storybook/react";
+import ComponentInfo from "../components/component-info";
+import packageConfig from "../../packages/wonder-blocks-form/package.json";
+import {NavigationTabItem} from "@khanacademy/wonder-blocks-tabs";
+import Link from "@khanacademy/wonder-blocks-link";
+
+/**
+ * A component for a tab item in NavigationTabs. It is used with a Link
+ * component.
+ */
```
Comment (jandrade) [inline __docs__/wonder-blocks-tabs/navigation-tab-item.stories.tsx:11]: **thought:** This comment block could probably be included in the code implementation directly, so devs could see it when using intellisense in their IDEs. (I'm not sure if there's a limitation with Storybook).

Same question applies to the `NavigationTabs` comment block.

**Review by jandrade — COMMENTED** [review]
Taking a first pass on this, this is taking great shape! Thanks for adding a11y unit tests from the start 👏

File: packages/wonder-blocks-tabs/src/components/navigation-tab-item.tsx [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        paddingInline: 0,
+        position: "relative",
+        ":hover": {
+            textDecoration: "none",
+            [":after" as any]: {
+                ...underlineStyles,
+                backgroundColor:
+                    semanticColor.action.primary.progressive.hover.border,
+            },
+        },
```
[the remark this answers is not in the document]
Comment (beaesguerra) [reply packages/wonder-blocks-tabs/src/components/navigation-tab-item.tsx:57]: Great question! I didn't know `textUnderlineOffset` and `textDecorationThickness` were a thing! Gave it a try and the underline is only applied under the text and not the icons unfortunately! I'll add a comment in the styles too so we know why we don't use `textDecoration` instead!

Open to suggestions though if you have ideas!

image

File: packages/wonder-blocks-tabs/src/components/navigation-tab-item.tsx [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    },
```
[the remark this answers is not in the document]
Comment (beaesguerra) [reply packages/wonder-blocks-tabs/src/components/navigation-tab-item.tsx:58]: Thanks for noticing this! I will iterate on the styles in another PR to address other cases and will add that then!

File: __docs__/wonder-blocks-tabs/navigation-tab-item.stories.tsx (hunk 3 above)
Comment (beaesguerra) [reply __docs__/wonder-blocks-tabs/navigation-tab-item.stories.tsx:11]: Good call! I added docs to the component js docs too so it shows up using intellisense (with an example 😄)! 

image

I kept this block around so it also shows in the Storybook docs page!

Comment (beaesguerra) [reply __docs__/wonder-blocks-tabs/navigation-tab-item.stories.tsx]: Update: Nevermind! Storybook extracts the component docs from the jsdocs! We can let Storybook generate the description for the component :)

File: packages/wonder-blocks-tabs/src/components/navigation-tab-item.tsx [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+        position: "relative",
+        ":hover": {
+            textDecoration: "none",
+            [":after" as any]: {
+                // We use :after to apply underline styles instead of textDecoration
+                // so the underline is shown under icons in Link too
+                ...underlineStyles,
+                backgroundColor:
+                    semanticColor.action.primary.progressive.hover.border,
+            },
```
Comment (jandrade) [inline packages/wonder-blocks-tabs/src/components/navigation-tab-item.tsx:78]: I'd like to propose another approach to get your thoughts on:

Do you think `boxShadow` could work in this case?

e.g.
```
boxShadow: inset 0 -${sizing.size_050} 0 0 ${semanticColor.action.primary.progressive.hover.border}
```

If you think that works, this could simplify the `:after` approach. Also would play nice with the idea we have for changing global focus to use `box-shadow` instead of `outline` :)

Comment (jandrade) [reply packages/wonder-blocks-tabs/src/components/navigation-tab-item.tsx:78]: @beaesguerra After discussing offline, we have found that this needs to be revisited in the next PR (or the PR that tackles styling). Probably `boxShadow` would still be suitable based on the latest design specs.

**Review by jandrade — APPROVED** [review]
Looks good to me! looking forward to see more progress on this component 🚢
